### ISSUE TITLE
Replace Travis CI with GitHub Actions

### ISSUE THREAD

**Issue by safris** [opentracing-contrib/java-specialagent#312]
GitHub actions [offers more concurrent builds](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#usage-limits) than can be achieved with Travis, which only provides 5.

**Comment by safris** [comment]
After an exceedingly lengthy time in attempting to achieve this enhancement, I have decided to keep SpecialAgent's build pipeline on Travis CI.

The reason for migrating to GitHub Actions was primarily to speed up the build time, because GitHub Actions allows 20 concurrent jobs instead of the 5 that Travis CI allows.

**Problems with Travis CI**

1. Travis CI allows only 5 concurrent jobs, which results in 1hr+ build times.
1. Travis CI is unstable, resulting in VMs failing to initialize, or other one-off errors during test execution.
1. The job VMs in Travis CI sometimes take a really long time to start (have to wait for hours sometimes).

**Problems with GitHub Actions**

1. GitHub Actions does not provide Oracle JDK java VMs.
1. GitHub Actions only provides Ubuntu Bionic and Xenial, but SpecialAgent needs Ubuntu Trusty.
1. The GitHub Actions log is super buggy!
1. Maven deployments from GitHub Actions (running on either Bionic or Xenial) always fail with timeout exception. This is an elusive issue that dates back to Maven 3.5.4, and I actually filed [an issue with OSS Sonatype in October 2018](https://issues.sonatype.org/browse/OSSRH-43371). This issue is occurring again for Maven 3.6.x, which is what GitHub Actions installs by default. I have figured out a way to install earlier versions of Maven, but this does not seem to fix the issue. It is most likely also related to the OS version, which cannot be changed with GitHub Actions.

The work for this issue is present [here](https://github.com/opentracing-contrib/java-specialagent/tree/github-actions).
